### ISSUE TITLE
Product Collection 

### ISSUE THREAD

**Issue by kmanijak** [woothemes/woocommerce#42389]
## Description
This epic covers a work to create a **Product Collection** block with the functionality of **Products (Beta)** block but built as a standalone/forked block, not as a variation of Query Loop. That includes foundation work to cover the current Products block functionalities, improvements around Inspector Controls compared to the current Products block, and a set of basic patterns for a new block.

Designs: `AtfOeZIFizx7ASmoSUbNjD-fi-55-9579&t=rnEPoeIr070VybIn-0`

## Technical guidelines/requirements
- General:
  - Product Collection is a standalone block and NOT a variation
  - Product Collection implementation can be based on Query Loop and Post Template blocks implementation
  - Product Collection should reuse as much of the existing code and must use Product Elements
- Frontend:
  - Frontend has to be as performant as Products block is (that requires PHP SSR on the frontend)
  - Consider using Interactivity API (TBD)
- Editor: 
  - Inspector Controls should be built with the new structure in mind from the very beginning
  - There still should be an option to inherit the global query

---
Section added 17 Jul 2023:

Here are the remaining issues split into must-do before release and post-release (not all of issues are mentioned here, only the high priority ones).

BEFORE releasing first version (SP: 20 + PHP Unit Testing):
- https://github.com/woocommerce/woocommerce-blocks/issues/10176
- https://github.com/woocommerce/woocommerce-blocks/issues/9783
- https://github.com/woocommerce/woocommerce/issues/42301
- https://github.com/woocommerce/woocommerce-blocks/issues/9703
- https://github.com/woocommerce/woocommerce-blocks/issues/10194 + potential fixes
- https://github.com/woocommerce/woocommerce-blocks/issues/10232
- https://github.com/woocommerce/woocommerce-blocks/issues/10230
- https://github.com/woocommerce/woocommerce-blocks/issues/10228
- https://github.com/woocommerce/woocommerce-blocks/issues/10328

ROLLOUT:
- https://github.com/woocommerce/woocommerce-blocks/issues/10430
- https://github.com/woocommerce/woocommerce/issues/42270

AFTER releasing first version (SP: 17):
- Data flow:
    - https://github.com/woocommerce/woocommerce-blocks/discussions/9993
    - https://github.com/woocommerce/woocommerce-blocks/issues/9268
    - https://github.com/woocommerce/woocommerce-blocks/pull/9469
- Features:
    - https://github.com/woocommerce/woocommerce/issues/42377
    - https://github.com/woocommerce/woocommerce-blocks/issues/9971
- Patterns:
    - https://github.com/woocommerce/woocommerce-blocks/issues/9372
    - https://github.com/woocommerce/woocommerce-blocks/issues/9368
    - https://github.com/woocommerce/woocommerce-blocks/issues/9369
    - https://github.com/woocommerce/woocommerce-blocks/issues/9370
    - https://github.com/woocommerce/woocommerce-blocks/issues/9371
---

## Phases
Here are the next phases of the work with the milestone that describes the end goal of each phase. Phases don't have to be addressed one after another but can overlap with each other. In parallel to the following phases, there will be additional work that still contributes to the Products and Product Collection blocks like bug fixes or Product Element improvements.

### 1. Exploratory

Foundation work with POC of the block. that includes preparing a base for a block by _forking_ the Query Loop and Post Template and adjusting the logic of the block to be product-centric (e.g. no data over-fetching in Editor etc.)

**Milestone:** Recognize the technical capabilities, limitations, and potential risks, and provide a block foundation.

Issues:
- https://github.com/woocommerce/woocommerce-blocks/issues/9267
- https://github.com/woocommerce/woocommerce-blocks/issues/9268
- https://github.com/woocommerce/woocommerce/issues/42346
- https://github.com/woocommerce/woocommerce/issues/42345
 
### 2. MVP/Beta

Cover the current Products functionality, especially assuring the compatibility of existing settings (like filters in both editor and frontend) or the ability to inherit the global query.

**Milestone:** Make the Product Collection as usable as Products block.

Issues:
- https://github.com/woocommerce/woocommerce-blocks/issues/9356
- https://github.com/woocommerce/woocommerce-blocks/issues/9358
- https://github.com/woocommerce/woocommerce-blocks/issues/9359
- Existing filters:
    - https://github.com/woocommerce/woocommerce-blocks/issues/9361
    - https://github.com/woocommerce/woocommerce-blocks/issues/9362
    - https://github.com/woocommerce/woocommerce-blocks/issues/9363
    - https://github.com/woocommerce/woocommerce-blocks/issues/9364
    - https://github.com/woocommerce/woocommerce-blocks/issues/9365
    - https://github.com/woocommerce/woocommerce-blocks/issues/9722
- ✅ Global Styles support - Product Collection supports the same set and scope of Global Styles as Products
- https://github.com/woocommerce/woocommerce-blocks/issues/9876

### 3. Editor: Inspector Controls improvements

Add additional sorting and filtering option.

**Milestone:** Make Product Collection fully configurable on the Editor side and make Inspector Controls more user-friendly.

Issues:
- https://github.com/woocommerce/woocommerce/issues/42377
- https://github.com/woocommerce/woocommerce-blocks/issues/9366
- https://github.com/woocommerce/woocommerce-blocks/issues/9367

### 4. Patterns

Implement a set of basic patterns for Product Collection.

**Milestone:** User can choose one of the patterns for most common use cases with the preset configuration.

Issues:
- https://github.com/woocommerce/woocommerce-blocks/issues/9372
- https://github.com/woocommerce/woocommerce-blocks/issues/9368
- https://github.com/woocommerce/woocommerce-blocks/issues/9369
- https://github.com/woocommerce/woocommerce-blocks/issues/9370
- https://github.com/woocommerce/woocommerce-blocks/issues/9371

### 5. Remaining work from Products block

There's work that contributes to the both: Products block and Product Collection. This work is a parallel to all to other phases.

Issues:
- https://github.com/woocommerce/woocommerce-blocks/issues/9242
- https://github.com/woocommerce/woocommerce/issues/42409
- https://github.com/woocommerce/woocommerce/issues/42415
- https://github.com/woocommerce/woocommerce/issues/42426
- https://github.com/woocommerce/woocommerce/issues/42487
- https://github.com/woocommerce/woocommerce/issues/42438
- https://github.com/woocommerce/woocommerce/issues/42351

### 6. Misc

- Documentation
- https://github.com/woocommerce/woocommerce-blocks/issues/10176
- https://github.com/woocommerce/woocommerce-blocks/issues/9783
- https://github.com/woocommerce/woocommerce/issues/42301
- https://github.com/woocommerce/woocommerce-blocks/issues/9703
- https://github.com/woocommerce/woocommerce-blocks/issues/10178
- https://github.com/woocommerce/woocommerce-blocks/issues/10194

### 7. Extras

Additional work extending the basic scope, feature requests etc.

Issues:
- https://github.com/woocommerce/woocommerce-blocks/issues/9971
- https://github.com/woocommerce/woocommerce-blocks/issues/10138

### 8. Bugs

- https://github.com/woocommerce/woocommerce-blocks/issues/9999

```[tasklist]
### Tasks
```

**Comment by kmanijak** [comment]
Closing in favor of other epics (https://github.com/woocommerce/woocommerce/issues/42161). This one is not used anymore
